src/main/java/org/xbill/DNS/SimpleResolver.java

```diff
@@ -444,4 +426,90 @@ private Message sendAXFR(Message query) throws IOException {
   public String toString() {
     return "SimpleResolver [" + address + "]";
   }
+
+  private class VerifyOneResponse {
```

Comment:
Actually, this doesn't need to be a class. As I mentioned before, throwing the exceptions is fine and whole thing can be a method: `Message verifyResponse(Message query, byte[] in) throws WireParseException { ... }`